src/mixture_laws.jl

```diff
@@ -95,7 +95,7 @@ extinction and is valid over [2, 6]. A typical value for the Milky Way is 3.1.
     f_A = 1.0
 end
 
-G16(Rv, f_A) = G16(promote(Rv, f_A)...)
+#G16(Rv, f_A) = G16(promote(Rv, f_A)...)
 #G16(Rv=3.1, f_A=3.1) = G16(Rv, f_A)
```

Comment:
I was also curious about what this line does. Is it a multiple dispatch thing?

Reply:
`promote` is a complex mechanism but it ensures that the outputs are all of the same type. To get a rough idea, look at the following examples
```julia
promote(1, 1.0)
promote(2.0, 3.5)
promote(3f0, 1)
promote(3f0, 1e2)
promote("1", 1)
```

it finds the "lowest common denominator" type for all inputs. A common design pattern is to type-def a struct such as 
```julia
struct TrivialStruct{T}
	a::T
	b::T
end
```
which means both fields `a` and `b` are of type `T` (look up parametric structs for more info). This means if I call `Trivial
Struct(1, 1.0)` it fails, because `1` and `1.0` do not have the same type. If I call `TrivialStruct(promote(1, 1.0)...)` it will promote the two inputs to the closest common type and use that as the input for the struct. None of this is required: it's just common practice for parametric structs!

Reply:
Right, thanks Miles. I appreciate the examples! I just realized I the section I highlighted could look misleading. My question was more about the uncommented bit directly below.

```julia
G16(Rv=3.1, f_A=3.1) = G16(Rv, f_A)
```

Unless that is what you meant and it is also a promotion thing? Sorry, still getting used to the new syntax!

Reply:
Oh sorry I misunderstood. That line is a function definition. We're using it to set defaults for a constructor. We do this because the with_kw macro makes a keyword argument only method.

Reply:
Oh, that's pretty neat. Thank!